philips_hue_test.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for functions in philips_hue.py."""
+
+import pytest
+
+import main
+import philips_hue
+import philips_hue_mock
+
+
+@pytest.fixture
+def config():
+    main.app.config.from_object('config.DevConfig')
+    return main.app.config
+
+
+@pytest.fixture
+def philips_hue_client(config):
+    philips_hue_client = philips_hue.PhilipsHueClient(
+        config['BRIDGE_IP_ADDRESS'], config['USERNAME'])
```

Comment:
Can we define constants in this test file, instead of loading an environment variable via config.py. For two reasons:

1. Isolate the code under test. This module should test philips_hue.py. Failures in config.py shouldn't trigger cascading unit test failures throughout the project.
2. Make it reproducible. Running on different machines (with different environment variables loaded) might cause the test to execute differently. We want to avoid depending on variable inputs when possible (another common example: define a time constant instead of calling datetime.datetime.now()), because when the tests fails, it might be impossible to reproduce that failure, because the input will have changed by the time we get around to investigating.

In this case, I don't think that we need to import main at all. I would do something like:

```
_BRIDGE_IP_ADDRESS = # Create a fake address, I forget the format offhand
_USERNAME = 'some-test-user'
_LIGHT_ID = 1

@pytest.fixture
def philips_hue_client(config):
    return philips_hue.PhilipsHueClient(_BRIDGE_IP_ADDRESS, _USERNAME)
```

Again, note the underscores, which indicates that we intend for these to be "private" variables (quotation marks because Python doesn't have private variables, this is just a convention that Python programmers follow).

Reply:
Done. I also decided to define the constants in the config fixture.